cwms-data-api/src/main/java/cwms/cda/data/dto/watersupply/WaterSupplyAccounting.java

```diff
@@ -0,0 +1,96 @@
+/*
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Hydrologic Engineering Center
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.data.dto.watersupply;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import cwms.cda.data.dto.CwmsDTOBase;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.annotations.FormattableWith;
+import cwms.cda.formatters.json.JsonV1;
+import java.util.List;
+
+@FormattableWith(contentType = Formats.JSONV1, formatter = JsonV1.class,
+        aliases = {Formats.DEFAULT, Formats.JSON})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonDeserialize(builder = WaterSupplyAccounting.Builder.class)
+@FormattableWith(contentType = Formats.JSONV1, formatter = JsonV1.class)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public final class WaterSupplyAccounting extends CwmsDTOBase {
+    @JsonProperty(required = true)
+    private final String contractName;
+    @JsonProperty(required = true)
+    private final WaterUser waterUser;
+    private final List<PumpAccounting> pumpAccounting;
```

Comment:
This data structure should be better defined. It should account for the different pumps and then the accounting times for a given pump. It should be able to easily answer without doing a full list scan whether a given pump has accounting, and what the given pump's start and end times are for the accounting.

Reply:
Restructured for less repeated data and better scaling